### DISCUSSION TITLE
Device Selection on Racks Page

### DISCUSSION THREAD

**Discussion by mike-andretta** [netbox-community/netbox#12698] [Help Wanted!] [answered]
### NetBox version

v3.5.1

### Python version

3.9

### Steps to Reproduce

1.) Click on Orginization
2.) Select Racks
3.) Select Rack name
4.) Click on Device in Rack on right hand side

### Expected Behavior

Device page should open.

### Observed Behavior

Device page should open, it does not. It looks like when a site is running HTTPS the redirect on the device in the rack it sent to an HTTP page, not an HTTPS page. This would break for anyone not supporting HTTPS on their netbox deployment. 

Link that Rack Device selection redirects to:  **_http://netbox.domain.com/dcim/devices/2/_**
Link that Rack Device selection should redirect to: **_https://netbox.domain.com/dcim/devices/2/_**

**Comment by PaulR282** [comment]
I can not reproduce this in v3.5.2. Just working fine, everything HTTPS

**Comment by mike-andretta** [comment]
> I can not reproduce this in v3.5.2. Just working fine, everything HTTPS

Is this an issue only in v3.5.1?

**Comment by PaulR282** [comment]
> Is this an issue only in v3.5.1?

Nope, I also can't reproduce it in v3.5.1. It's something with your setup.

**Comment by mike-andretta** [comment]
Great, cant be reproduced, this is a brand new setup. What would be causing this, where can one look to see something in their setup that would make this happen only on the racks page?

**Comment by candlerb** [comment, marked as answer]
Almost certainly you haven't set the `X-Forwarded-Proto:` header on your proxy.

There are [sample files](https://github.com/netbox-community/netbox/tree/develop/contrib) for nginx and apache you can copy which do this properly.

**Reply by mike-andretta** [reply]
That was it, it was this line that needed to be added to my config. 

RequestHeader set "X-Forwarded-Proto" expr=%{REQUEST_SCHEME}
